service/state/core_access.go

```diff
@@ -157,6 +160,15 @@ func (ca *CoreAccessor) BalanceForAddress(ctx context.Context, addr Address) (*B
 	}
 	// unmarshal balance information
 	value := result.Response.Value
+	// if the value returned is empty, the account balance does not yet exist
+	if len(value) == 0 {
+		log.Errorf("balance for account %s does not exist at block height %d, "+
+			"wait for node to sync", addr.String(), head.Height)
```

Comment:
Should we not keep this suggestion? It is not going to be actual once we merge Syncer

Reply:
Yes, how likely is it going to be that a user is requesting balance of acct that doesn't exist in trie?

Reply:
I mean specifically about the `wait for node to sync"`

Reply:
The assumption that I'm referring to here is that a user would *likely* not request the balance of an account that does not yet exist in acct trie, meaning the node should sync up to network head where some acct data exists, does that make sense?